cdbot/cogs/maths.py

```diff
@@ -144,15 +144,26 @@ async def challenge(self, ctx: Context, number: int = 1):
         return await ctx.send(embed=embed)
 
     @command()
-    async def latex(self, ctx: Context, *, expression: str):
+    async def latex(self, ctx: Context):
         """
         Render a LaTeX expression with https://quicklatex.com/
         """
-        await self.latex_render(ctx, expression)
+        await self.latex_render(ctx, ctx.message)
 
-    async def latex_render(self, ctx: Context, expression: str):
+    async def latex_render(self, ctx: Context, message: Message):
+
+        # When invoked with the word latex, it is run as a command so ctx is type Context
+        # but when invoked with just $$$$ the context is a TextChannel
+        # so we need to check ctx's type
         channel = ctx.channel.id if type(ctx) is Context else ctx.id
 
+        # get the latex expression to be rendered
+        expression = message.content
+        if message.content.startswith("...latex "):
```

Comment:
I know we did this before but :vomiting_face:, a better solution should be found. @AnonGuy thoughts?

Reply:
Is it really that bad? The message needs to be passed to the function when inline or it can't be stored for checking anyway, so passing in the message as well as the message content seems redundant. And about the if, that can be made a one liner if you want (there's a similar if below so I thought you guys were ok with it). `...latex \LaTeX` still works if you want a gif of that... I might be a noob but is there a particular reason this is bad code?

Reply:
passing the expression to `latex_render` as well as the message does seem like a good way to to it though, it you really don't like this.